docs/compiler-exceptions.rst

```diff
@@ -0,0 +1,276 @@
+
+Compiler Exceptions
+*******************
+
+.. _exceptions-common::
+
+Compiler Exceptions
+===================
+
+These are examples of the compiler exception errors that you could see when
+compiling a ``vyper`` file for deployment on the Ethereum Virtual Machine.
+
+
+.. py:exception:: ArrayIndexException
+
+    This exception will occur when an invalid index number of an array is referenced.  The terminal will point out the line 
+    which contains the error.
+
+.. py:exception:: ConstancyViolationException
+
+    This exception occurs when a variable or function that is returning a constant has another instance that is trying
+    to change the value.
+
+.. py:exception:: EventDeclarationException
+
+    This exception will occur when the identifier used to declare an event is in conflict with a reserved word
+    or previously declared structure with the same name.  The terminal output will state the line which contains the error.
+
+.. py:exception:: EMVVersionException
+
+    .. code-block:: python3
+
+        {
+            "settings": {
+                "evmVersion": "[VERSION]"
+            }
+        }
+
+    Default version is ``istanbul``.  Other version choices include ``byzantium``, ``constantinople``, and ``petersburg``.  This
+    exception will occur when the compiler version is not compatible with the EVM version declared in the code.
+
+.. py:exception:: FunctionDeclarationException
+
+    .. code-block:: python3
+
+        struct Bid:
+            blindedBid: bytes32
+            deposit: wei_value
+        @public
+        @payable
+            def pendingReturns(_blindedBid: bytes32):
+        # Check if bidding period is still open
+        assert block.timestamp < self.biddingEnd
+
+        # Check that payer hasn't already placed maximum number of bids
+        numBids: int128 = self.bidCounts[msg.sender]
+        assert numBids < MAX_BIDS
+
+        # Add bid to mapping of all bids
+        self.bids[msg.sender][numBids] = Bid({
+            blindedBid: _blindedBid,
+            deposit: msg.value
+        })
+            self.bidCounts[msg.sender] += 1
+    
+        pendingReturns: map(address, wei_value)
+
+    ``FunctionDeclarationException`` happens when a function name is used for two different functions or when a reserved word 
+    is used to name a function.
+
+.. code-block:: bash
+
+    $ vyper blind_auction.vy
+    Error compiling: blind_auction.vy
+    /usr/lib/python3/dist-packages/apport/report.py:13: DeprecationWarning: the imp module is deprecated in favour of             importlib; see the module's documentation for alternative uses
+    import fnmatch, glob, traceback, errno, sys, atexit, locale, imp
+    vyper.exceptions.FunctionDeclarationException: Function name shadowing a variable name: pendingReturns
+
+The warning generated in the terminal does not specify any line numbers.  ``pendingReturns`` is named as the identifier
```

Comment:
Missing indent on this line and the following one.

Reply:
Fixed